From 0161f863081006784d348a0a9e6f02fcaddb3cc6 Mon Sep 17 00:00:00 2001
From: Witold Sadowski <wsadowski@marvell.com>
Date: Wed, 17 Mar 2021 06:54:41 -0700
Subject: [PATCH 1333/1921] spi: Disable configuration during simulation

During simulation some checks and trainings should not
be performed.
Allow to skip PHY training, or command check.

Change-Id: Ia98d63f367af2d66f8f980e41fcbf4342f127ad5
Signed-off-by: Witold Sadowski <wsadowski@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/47936
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/48122
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/spi/spi-cadence-xspi-pci.c | 1 +
 drivers/spi/spi-cadence-xspi.c     | 4 ++++
 drivers/spi/spi-cadence-xspi.h     | 1 +
 3 files changed, 6 insertions(+)

diff --git a/drivers/spi/spi-cadence-xspi-pci.c b/drivers/spi/spi-cadence-xspi-pci.c
index 3cc46a48c4e4..485b51f17142 100644
--- a/drivers/spi/spi-cadence-xspi-pci.c
+++ b/drivers/spi/spi-cadence-xspi-pci.c
@@ -56,6 +56,7 @@ static int cadence_octeon_spi_probe(struct pci_dev *pdev,
 	cdns_xspi->auxbase  = register_base + AUX_OFFSET;
 	cdns_xspi->sdmabase = register_base + STIG_OFFSET;
 	cdns_xspi->irq = 0;
+	cdns_xspi->skip_sim_check = true;
 
 	init_completion(&cdns_xspi->cmd_complete);
 	init_completion(&cdns_xspi->sdma_complete);
diff --git a/drivers/spi/spi-cadence-xspi.c b/drivers/spi/spi-cadence-xspi.c
index 7b11db5071d9..4bed78952be7 100644
--- a/drivers/spi/spi-cadence-xspi.c
+++ b/drivers/spi/spi-cadence-xspi.c
@@ -62,6 +62,8 @@ static int cdns_xspi_read_dqs_delay_training(struct cdns_xspi_dev *cdns_xspi)
 	}
 
 	if (rd_dqs_del_min == -1) {
+		if (cdns_xspi->skip_sim_check)
+			return 0;
 		dev_err(cdns_xspi->dev, "PHY training failed\n");
 		return -EBUSY;
 	} else if (rd_dqs_del_max == -1) {
@@ -199,6 +201,8 @@ static int cdns_xspi_check_command_status(struct cdns_xspi_dev *cdns_xspi)
 			}
 		}
 	} else {
+		if (cdns_xspi->skip_sim_check)
+			return 0;
 		dev_err(cdns_xspi->dev, "Fatal error - command not completed\n");
 		ret = -EPROTO;
 	}
diff --git a/drivers/spi/spi-cadence-xspi.h b/drivers/spi/spi-cadence-xspi.h
index 9e506ef1b039..1d5bcca4641f 100644
--- a/drivers/spi/spi-cadence-xspi.h
+++ b/drivers/spi/spi-cadence-xspi.h
@@ -235,6 +235,7 @@ struct cdns_xspi_dev {
 
 	int irq;
 	int current_cs;
+	bool skip_sim_check;
 
 	struct mutex lock;
 
-- 
2.31.1

